src/OpenAi.php

```diff
@@ -93,25 +125,52 @@ public function engine($engine)
      */
     private function sendRequest(string $url, string $method, array $opts = null)
     {
-        $curl_info = [
-            CURLOPT_URL => $url,
-            CURLOPT_RETURNTRANSFER => true,
-            CURLOPT_ENCODING => '',
-            CURLOPT_MAXREDIRS => 10,
-            CURLOPT_TIMEOUT => 0,
-            CURLOPT_FOLLOWLOCATION => true,
-            CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
-            CURLOPT_CUSTOMREQUEST => $method,
-            CURLOPT_POSTFIELDS => json_encode($opts),
-            CURLOPT_HTTPHEADER => $this->headers,
-        ];
+        $curl_info = [];
+        if($method == 'POST' || $method == 'GET') //custom post or get
+        {
+            $curl_info = [
+                CURLOPT_URL => $url,
+                CURLOPT_RETURNTRANSFER => true,
+                CURLOPT_ENCODING => '',
+                CURLOPT_MAXREDIRS => 10,
+                CURLOPT_TIMEOUT => 0,
+                CURLOPT_FOLLOWLOCATION => true,
+                CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+                CURLOPT_CUSTOMREQUEST => $method,
+                CURLOPT_POSTFIELDS => json_encode($opts),
+                CURLOPT_HTTPHEADER => $this->headers,
+            ];
+        }
+        elseif($method == 'FILE_POST') //case for the file[default post command]
```

Comment:
`$curl_info` is already defined, Thus defining many times isn't the best practice, You can unset unnecessary elements from the array according to the `$method` equalization condition.